### PR TITLE
KAFKA-3266: List/Alter Acls - protocol and server side implemen…

### PR DESCRIPTION
**Pull request by granthenke**
…tation

### CODE REVIEW

**Comment by granthenke** [comment]
I am still working on this patch. However it is close enough that I thought putting it up for some early reviews and feedback would be valuable. The List/Alter Acls changes for KIP-4 don't have the same blocking concerns as the rest of the messages since the Authorizer implementation is responsible for that.

**Comment by granthenke** [comment]
This patch has the test (commented out) and logging lines from [KAFKA-3328](https://issues.apache.org/jira/browse/KAFKA-3328): SimpleAclAuthorizer can lose ACLs with frequent add/remove calls

**Comment by granthenke** [comment]
@ijuma @junrao @gwenshap Pinging for review. I have updated the KIP-4 wiki with the details around these requests/response [here](https://cwiki.apache.org/confluence/display/KAFKA/KIP-4+-+Command+line+and+centralized+administrative+operations#KIP-4-Commandlineandcentralizedadministrativeoperations-ACLAdminSchema).

File: core/src/main/scala/kafka/server/KafkaApis.scala [hunk 1]
```diff
@@ -993,6 +994,112 @@ class KafkaApis(val requestChannel: RequestChannel,
...
+    val responseBody =
+      authorizer match {
+        case None =>
+          error("Error processing list acls request. No Authorizer is available.")
+          new ListAclsResponse(Map[JResource, JSet[JAcl]]().asJava, Errors.CLUSTER_AUTHORIZATION_FAILED)
+        case Some(auth) =>
+          // authorized if the principal has all access on the cluster
+          // or the principal is requesting their own acls
+          val isAuthorized = authorize(request.session, All, Resource.ClusterResource)
+          val isUserPrinciple = listAclsRequest.hasPrincipal && listAclsRequest.getPrincipal == request.session.principal
```
Comment (SinghAsDev) [inline core/src/main/scala/kafka/server/KafkaApis.scala:1010]: Typo?

**Comment by ijuma** [comment]
This work was completed via a different PR, closing.

**Pull request closed without merging**